zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java

```diff
@@ -1,98 +1,202 @@
+
+    // Segment input variables
+    private static final String ADHOC_SEGMENT = "/adhocs";
+    private static final String ARTICLE_SEGMENT = "/articles";
+    private static final String BULLETIN_SEGMENT = "/bulletins";
+    private static final String DATA_SEGMENT_WITHOUT_SLASH = "data";
+    private static final String DATASETS_SEGMENT = "/datasets";
+    private static final String LINE_CHART_CONFIG_SEGMENT = "/linechartconfig";
+    private static final String QMIS_SEGMENT = "/qmis";
+    private static final String TIME_SERIES_SEGMENT = "/timeseries";
+    private static final String VISUALISATIONS_SEGMENT = "/visualisations";
+
+    // Resource types
+    private static final List<String> RESOURCE_TYPE_SEGMENTS = Arrays.asList(
+            "/chartconfig?",
+            "/chartimage?",
+            "/embed?",
+            "/chart?",
+            "/resource?",
+            "/generator?",
+            "/file?",
+            "/export?"
+    );
+
+    private Http http;
 
     static {
         websiteHosts = Configuration.getWebsiteHosts();
     }
 
     public PublishNotification(Collection collection, List<String> urisToUpdate, List<ContentDetail> urisToDelete) {
+        this.collectionId = collection.getDescription().getId();
 
         // Delay the clearing of the cache after publish to minimise load on the server while publishing.
         Date clearCacheDate = new DateTime(collection.getDescription().getPublishDate())
                 .plusSeconds(Configuration.getSecondsToCacheAfterScheduledPublish()).toDate();
 
-        this.payload = new NotificationPayload(collection.getDescription().getId(), urisToUpdate, urisToDelete,
-                clearCacheDate);
+        if (Configuration.isLegacyCacheAPIEnabled()) {
+            if (urisToUpdate != null) {
+                for (String uriToUpdate : urisToUpdate) {
+                    NotificationPayloadCacheApi notificationPayloadCacheApi = new NotificationPayloadCacheApi(collection.getDescription().getId(), uriToUpdate, clearCacheDate);
+                    payloads.add(notificationPayloadCacheApi);
+                }
+            } else {
+                NotificationPayloadCacheApi notificationPayloadCacheApi = new NotificationPayloadCacheApi(collection.getDescription().getId(), null, clearCacheDate);
+                payloads.add(notificationPayloadCacheApi);
+            }
+        } else {
+            this.payload = new NotificationPayload(collection.getDescription().getId(), urisToUpdate, urisToDelete,
+                    clearCacheDate);
+        }
+
+    }
+
+    protected List<NotificationPayloadCacheApi> getPayloads() {
+        return this.payloads;
+    }
+
+    protected NotificationPayload getPayload() {
+        return this.payload;
     }
 
     public PublishNotification(Collection collection) {
         this(collection, null, null);
     }
 
+
+    public void setHttp(Http http) {
+        this.http = http;
+    }
+
+    private Http getHttp() {
```

Comment:
I think you have done this so that you can test it, but I think this is now potentially buggy. For instance, this is now a singleton, which can cause problems: for instance, if you check the implementation of `Http`, you'll see that it has a `headers` list. Every time you call `http.put`, you will combine the existing headers with the new ones you're passing. I think this works with the current implementation because we're always sending the same headers, but it's a potential problem.

Reply:
removed this method and created a class  SendNotification for the legacy cache API. That way it can be tested better and our new code is in a separate class not all over the place in the current PubilshNotification class.